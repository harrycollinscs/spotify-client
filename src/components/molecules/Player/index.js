import React, { PureComponent } from "react";
import SpotifyPlayer from "react-spotify-web-playback";
import store from "../../../store";

const PlayerStyle = {
  activeColor: "#fff",
  bgColor: "rgb(0,0,0)",
  color: "#fff",
  loaderColor: "#fff",
  sliderColor: "#1db954",
  trackArtistColor: "#ccc",
  trackNameColor: "#fff",
  height: "50px",
};

class Player extends PureComponent {
  render() {
    const { contextUri } = this.props;
    const uri = contextUri || "spotify:artist:6HQYnRM4OzToCYPpVBInuU";

    return (
      <SpotifyPlayer
        token={store.getState().accessToken}
        uris={[uri]}
        styles={PlayerStyle}
        autoPlay={false}
        syncExternalDevice={true}
        magnifySliderOnHover={true}
      />
    );
  }
}

export default Player;
