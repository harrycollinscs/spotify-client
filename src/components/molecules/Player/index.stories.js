import React from "react";
import Player from "./index";

export default {
  title: "Molecules/Player",
  component: Player,
};

const Template = (args) => <Player {...args} />;

export const Default = Template.bind({});
Default.args = {
  contextUri: "spotify:artist:6HQYnRM4OzToCYPpVBInuU",
};

