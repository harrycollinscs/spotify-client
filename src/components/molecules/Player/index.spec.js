import React from "react";
import Player from "./index";
import { shallow } from "enzyme";

describe("Molecules/Player", () => {
  const component = <Player />;

  it("Should match snapshot", () => {
    const render = shallow(component);
    expect(render).toMatchSnapshot();
  });
});
