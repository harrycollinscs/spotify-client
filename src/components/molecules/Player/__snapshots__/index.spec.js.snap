// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Molecules/Player Should match snapshot 1`] = `
<SpotifyWebPlayer
  autoPlay={false}
  callback={[Function]}
  magnifySliderOnHover={true}
  name="Spotify Web Player"
  showSaveIcon={false}
  styles={
    Object {
      "activeColor": "#fff",
      "bgColor": "rgb(0,0,0)",
      "color": "#fff",
      "height": "50px",
      "loaderColor": "#fff",
      "sliderColor": "#1db954",
      "trackArtistColor": "#ccc",
      "trackNameColor": "#fff",
    }
  }
  syncExternalDevice={true}
  syncExternalDeviceInterval={5}
  token={null}
  uris={
    Array [
      "spotify:artist:6HQYnRM4OzToCYPpVBInuU",
    ]
  }
/>
`;
